provider/oauth2/tests.py

```diff
@@ -113,22 +115,19 @@ def test_authorization_requires_response_type(self):
         self.assertEqual(400, response.status_code)
         self.assertTrue(escape(u"No 'response_type' supplied.") in response.content)
 
-    def test_authorization_requires_supported_response_type(self):
+    @ddt.data('code', 'token', 'unsupported')
+    def test_authorization_requires_supported_response_type(self, response_type):
         self.login()
         response = self.client.get(
-            self.auth_url(), self.get_auth_params(response_type="unsupported"))
+            self.auth_url(), self.get_auth_params(response_type=response_type))
         response = self.client.get(self.auth_url2())
 
-        self.assertEqual(400, response.status_code)
-        self.assertTrue(escape(u"'unsupported' is not a supported response type.") in response.content)
-
-        response = self.client.get(self.auth_url(), data=self.get_auth_params())
-        response = self.client.get(self.auth_url2())
-        self.assertEqual(200, response.status_code, response.content)
+        if response_type == 'unsupported':
+            self.assertEqual(400, response.status_code)
+            self.assertTrue(escape(u"'unsupported' is not a supported response type.") in response.content)
```

Comment:
i just moved that while refactoring an existing test.   i don't know why it's needed, but it is used elsewhere when checking error message output (in several tests i have not touched)